skin/frontend/rwd/default/js/configurableswatches/product-media.js

```diff
@@ -118,74 +125,122 @@ var ConfigurableMediaImages = {
     getImageObject: function(productId, imageUrl) {
         var key = productId+'-'+imageUrl;
         if(!ConfigurableMediaImages.imageObjects[key]) {
-            var image = $j('<img />');
-            image.attr('src', imageUrl);
+            var image = document.createElement('img');
+            image.src = imageUrl;
             ConfigurableMediaImages.imageObjects[key] = image;
         }
         return ConfigurableMediaImages.imageObjects[key];
     },
 
-    updateImage: function(el) {
-        var select = $j(el);
-        var label = select.find('option:selected').attr('data-label');
+    updateImage(el) {
+        var select = el;
+        var label = select.options[select.selectedIndex].getAttribute('data-label');
         var productId = optionsPrice.productId; //get product ID from options price object
 
         //find all selected labels
-        var selectedLabels = new Array();
+        var selectedLabels = [];
 
-        $j('.product-options .super-attribute-select').each(function() {
-            var $option = $j(this);
-            if($option.val() != '') {
-                selectedLabels.push($option.find('option:selected').attr('data-label'));
+        var superAttributeSelects = document.querySelectorAll('.product-options .super-attribute-select');
+        superAttributeSelects.forEach(function(option) {
+            if (option.value !== '') {
+                selectedLabels.push(option.options[option.selectedIndex].getAttribute('data-label'));
             }
         });
 
         var swatchImageUrl = ConfigurableMediaImages.getSwatchImage(productId, label, selectedLabels);
-        if(!ConfigurableMediaImages.isValidImage(swatchImageUrl)) {
+        if (!ConfigurableMediaImages.isValidImage(swatchImageUrl)) {
+            console.log('no image found');
             return;
         }
 
         var swatchImage = ConfigurableMediaImages.getImageObject(productId, swatchImageUrl);
 
-        ProductMediaManager.swapImage(swatchImage);
+        this.swapImage(swatchImage);
+    },
+
+    swapImage: function(targetImage) {
```

Comment:
since this funcion is now here, do you think we could remove it from app.js or it's needed for some other place?

Reply:
function swapImage in app.js contains also controls of zoom plugin,
maybe it's better to control existence of ProductMediaManager object and add a condition here so it can work with and without app.js that include zoom and other things